### PR TITLE
Render compositions for different examples into different files

### PR DESCRIPTION
**Pull request by kaessert**
This PR tries to achieve two goals:
- save output of render into cache-folder
- save to separate files for separate languages 

This allows us to build further commands like `make render.test` or `make validate` without additional boiler-plate code

### CODE REVIEW

File: makelib/uptest.mk [hunk 1]
```diff
@@ -54,8 +54,24 @@ uptest: $(UPTEST) $(KUBECTL) $(CHAINSAW) $(CROSSPLANE_CLI) $(YQ)
 # Run uptest together with all dependencies. Use `make e2e UPTEST_SKIP_DELET=true` to skip deletion of resources.
 e2e: build controlplane.down controlplane.up $(UPTEST_LOCAL_DEPLOY_TARGET) uptest #
 
+# Renders crossplane compositions in the current project
+#
+# Compositioni and Function must be defined, Environment and Observeed Resources are optionally available.
```
Comment (haarchri) [inline makelib/uptest.mk:59]: nit: is this italian ? ;)

Comment (kaessert) [reply makelib/uptest.mk]: fixed :)

File: makelib/uptest.mk [hunk 2]
```diff
@@ -72,12 +88,18 @@ render: $(CROSSPLANE_CLI) ${YQ}
 			if [[ "$$COMPOSITION" == "null" || "$$FUNCTION" == "null" ]]; then \
 				continue; \
 			fi; \
+			LANG=$$(basename "$$(dirname "$$file")"); \
```
Comment (ytsarev) [inline makelib/uptest.mk:91]: That looks like not obvious assumption/convention. We need to document it explicitly

File: makelib/uptest.mk [hunk 3]
```diff
@@ -54,8 +54,24 @@ uptest: $(UPTEST) $(KUBECTL) $(CHAINSAW) $(CROSSPLANE_CLI) $(YQ)
...
 
+# Renders crossplane compositions in the current project
+#
+# Composition and Function must be defined, Environment and Observeed Resources are optionally available.
+# The command discovers sees by parsing the `render.crossplane.io/*`-annotations of file found in the
+# examples-directorry, usually: `/examples`
+#
+# Possible values are:
+#  - composition-path (Composition)
+#  - functino-path (Function)
```
Comment (ytsarev) [inline makelib/uptest.mk:65]:
```suggestion
#  - function-path (Function)
```

Comment (kaessert) [reply makelib/uptest.mk]: ty! fixed

**Review by ytsarev — APPROVED** [review]
LGTM

File: makelib/uptest.mk (hunk 2 above)
Comment (kaessert) [reply makelib/uptest.mk]: Yes you're right, it's about having separate files, no need to bring assumption about language into picture :)

**Review by sergenyalcin — APPROVED** [review]
Thanks @kaessert LGTM!

**Pull request merged**